openedx/core/djangoapps/olx_rest_api/block_serializer.py

```diff
@@ -0,0 +1,160 @@
+"""
+Code for serializing a modulestore XBlock to OLX suitable for import into
+Blockstore.
+"""
+import logging
+import os
+from collections import namedtuple
+
+from lxml import etree
+
+from . import adapters
+
+log = logging.getLogger(__name__)
+
+# A static file required by an XBlock
+StaticFile = namedtuple('StaticFile', ['name', 'url', 'data'])
+
+
+def blockstore_def_key_from_modulestore_usage_key(usage_key):
+    """
+    In modulestore, the "definition key" is a MongoDB ObjectID kept in split's
+    definitions table, which theoretically allows the same block to be used in
+    many places (each with a unique usage key). However, that functionality is
+    not exposed in Studio (other than via content libraries). So when we import
+    into Blockstore, we assume that each usage is unique, don't generate a usage
+    key, and create a new "definition key" from the original usage key.
+    So modulestore usage key
+        block-v1:A+B+C+type@html+block@introduction
+    will become Blockstore definition key
+        html/introduction
+    """
+    block_type = usage_key.block_type
+    if block_type == 'vertical':
+        # We transform <vertical> to <unit>
+        block_type = "unit"
+    return block_type + "/" + usage_key.block_id
+
+
+class XBlockSerializer(object):
+    """
+    This class will serialize an XBlock, producing:
+        (1) A new definition ID for use in Blockstore
+        (2) an XML string defining the XBlock and referencing the IDs of its
+            children (but not containing the actual XML of its children)
+        (3) a list of any static files required by the XBlock and their URL
+    """
+
+    def __init__(self, block):
+        """
+        Serialize an XBlock to an OLX string + supporting files, and store the
+        resulting data in this object.
+        """
+        self.orig_block_key = block.scope_ids.usage_id
+        self.static_files = []
+        self.def_id = blockstore_def_key_from_modulestore_usage_key(self.orig_block_key)
+
+        # Special cases:
+        if self.orig_block_key.block_type == 'html':
+            self.serialize_html_block(block)
+        else:
+            self.serialize_normal_block(block)
+
+        course_key = self.orig_block_key.course_key
+        # Search the OLX for references to files stored in the course's
+        # "Files & Uploads" (contentstore):
+        self.olx_str = adapters.rewrite_absolute_static_urls(self.olx_str, course_key)
+        for asset in adapters.collect_assets_from_text(self.olx_str, course_key):
+            #self.add_static_asset(asset['content'])
```

Comment:
Is this line for future reference?

Reply:
Whoops, no that's just old dead code. I'll delete that line.